### ISSUE TITLE
Tombo resquiggle "alignment not produced"

### ISSUE THREAD

**Issue by impreza-92** [nanoporetech/tombo#136]
Hello, 

When I try to run the resquiggle command i get the error "Alignment not produced" for 95% of my reads. In the manual it says that it means that minimap2/mappy cannot produce an alignment. Is there a way I can see why it does not produce an alignment? Whenever I use minimap2 by itself to align my reads to my reference it works just find and I get proper .sam/.bam files. Have you perhaps an idea why it might not properly align when doing a resquiggle? 

In short: Is there a way to get more information about the "Alignment not produced" error from the resquiggle command?

**Comment by marcus1487** [comment]
The mappings should be identical to the minimap2 command line mapping (assuming the `map-ont` preset setting). One other possibility is that the installed minimap2 and mappy versions are different.

As for the "events" table, I am not sure I understand this question. Tombo does not read (or overwrite) the base calling events table from the FAST5s (stored in the `Basecall_1D_000` slot). Tombo writes to a separate slot (`RawGenomeCorrected_000 ` by default) which would include events if the read was re-squiggled successfully or the status attribute with an error message otherwise.

**Comment by impreza-92** [comment]
I will try to see if I can align with mappy by itself to see if the issue is there. 
Like you explain, I do see the unsuccesful status in the RawGenomeCorrected_000. 

Is there perhaps a way to view the errors that mappy is throwing in Tombo? Again, when I just us minimap2 on my .fastq's and the reference, I get a .sam and I can visualize the alignment with IGV. If I understand correctly, during resquiggling the fastq information in the .fast5 files is used, right?

**Comment by marcus1487** [comment]
You have indeed found the correct bit of code where the "Alignment not produced" error is raised. The excepting of the StopIteration exception results from the `next` call in the above code. This indicates that no alignment was produced by the `aligner.map` call (i.e. there is no valid alignment for this read). So the sequence was successfully extracted from the FAST5 (in the line before the try block, but no valid alignment from the reference was found.

If reads from the FASTQ map well, my best guess is that the FASTQ slot in the FAST5s do not contain the same base calls. I would recommend running the `tombo preprocess annotate_raw_with_fastqs` command with the `--overwrite` flag to ensure that the appropriate base calls are found within the FAST5s.

**Comment by marcus1487** [comment]
Can it be confirmed that the FASTQs in the FAST5 files are the same as those referenced for external mapping?

**Comment by impreza-92** [comment]
Yes, they are them same. Though, after further digging, I have to admit that I seem to have a used a subset of reads of which the majority did map in external testing causing me to believe that the mapping was fine. However, when using ALL the reads in external testing I got the same results as during the resquiggle. So, to summarize: my reads do generally not map when using either minimap2 or Tombo.

On the other hand this leads me to a more overarching question.

The samples which we are discussing are ground truth methylation standards (C, 5mC, 5hmC, 5fC, 5caC). Upon investigating the alignments issues I also calculated the read errors for each sample (with a particular interest for C's). This showed that the read error for the "C" sample was ~10% which was to be expected. However, the other samples that included the modifications all had read errors of ~50%! Realtively a lot of the C's in these cases were called as deletions or T's. Furthermore, the other bases (A, G, T) also had ~50% error rates. I also found it typical that during resquiggling the "C" sample had only ~2.5% "Alignment not produced" whereas the other modified samples had ~95%. 

With the above in mind I had a few reads through the documentation again where it is stated that the 5mC and 6mA models were made with the Tombo itself (resquiggling etc.). I wonder, however, how well the resquiggling of these samples was (since modifications seem to introduce a lot of errors) and how that affects the integrity of the models produced. 

I think that the fact that ~95% of my reads have a lot of errors and therefore do not align during the resquiggle step is not good for making a solid model. 

Is it normal to see such high error rates when sequencing modified bases and did you not encounter this problem when making the current 5mC/6mA models?

**Comment by impreza-92** [comment]
Thank for you answer. It makes a lot more sense to me now! So if I understand correctly, you do not want to throw off the basecalling by introducing modifications too densely in a sequence? Does that in extension mean that sequences that are very densely methylated also cannot be used as a validation to test a new model (due to them not being able to be processed correctly by Tombo)? 

You mentioned that, ideally, you want to have 15%-35% modified bases of interest in your sequence but I wonder if you then have enough sequence context for the model to train on? Likely, it will not encounter modifications in all sequence contexts and therefore I wonder whether it can call modifications in unseen contexts. 

I will take your advice to heart and try to make different samples that will work better with Tombo.
